backend/controllers/roleAdminTypeAmendment.js

```diff
@@ -15,31 +20,34 @@ const roleAdminTypeAmendmentFactory = (getRoleDetailsApi, changeRoleAdminTypeApi
     const flashRole = req.flash('changeRoleAdminType')
     const roleAdminType = flashRole != null && flashRole.length > 0 ? flashRole[0] : roleDetails.adminType
 
-    // create map of all adminType and their status value
-    const adminTypeList = roleAdminType.map((e) => e.adminTypeCode)
-    const adminTypeMap = {
-      DPS_ADM: adminTypeList.includes('DPS_ADM'),
-      DPS_LSA: adminTypeList.includes('DPS_LSA'),
-      EXT_ADM: adminTypeList.includes('EXT_ADM'),
-    }
-
     res.render('changeRoleAdminType.njk', {
       title,
       roleUrl,
-      currentRoleAdminType: adminTypeMap,
+      adminTypeValues,
+      currentFilter: roleAdminType.map((e) => e.adminTypeCode),
       errors: req.flash('changeRoleErrors'),
     })
   }
 
+  const maintainImmutableAdminTypes = (originalAdminType, amendedAdminType) => {
+    const allImmutableAdminTypes = adminTypeValues.filter((item) => item.immutable === true).map((e) => e.value)
```

Comment:
Can the === true be dropped here?

Reply:
I think I tried that but it failed...will try again now all code is back

Reply:
It might not realise it's a boolean, no worries if it doesn't work

Reply:
Have checked and the boolean js check can be updated neatly - have pushed that fix back to this branch, but will need an additional review